core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/tags/SimpleTagQueryParser.java

```diff
@@ -301,6 +324,18 @@ public static boolean isRegExp(String tagValuesQuery) {
         return groupMetrics;
     }
 
+    private void logQuery(String tenantId, Query query, Set<? extends MetricId<?>> metricIds) {
+        // If debug is enabled, then always log the query info; otherwise, only log query info if the page threshold
+        // is exceeded so as to avoid spamming the log file. The page threshold is a simple mechanism to let us know
+        // that a particular query has a large result set and requires a number of round trips to Cassandra that exceeds
+        // the threshold.
+        if (logger.isDebugEnabled()) {
+            logger.debugf("Tag query %s returned %d rows", Query.toString(tenantId, query), metricIds.size());
+        } else if ((metricIds.size() / pageSize) > pageThreshold) {
+            logger.infof("Tag query %s returned %d rows", Query.toString(tenantId, query), metricIds.size());
```

Comment:
I assume this is supposed to indicate that the query fetched too many metrics? Shouldn't this really be a warning so that someone notices and might actually improve their query?